generator/internal/dart/dart.go

```diff
@@ -0,0 +1,309 @@
+	// TODO(codyoss): https://github.com/googleapis/google-cloud-rust/issues/34
+	for _, segment := range h.PathTemplate {
+		if segment.FieldPath != nil {
+			// TODO(#34) - handle nested path params
+			args = append(args, fmt.Sprintf(", req.%s", strcase.ToCamel(*segment.FieldPath)))
+		}
+	}
+	return args
+}
+
+func toPascal(symbol string) string {
+	return escapeKeyword(strcase.ToCamel(symbol))
+}
+
+func formatDocComments(documentation string, _ *api.APIState) []string {
+	ss := strings.Split(documentation, "\n")
+	for i := range ss {
+		ss[i] = strings.TrimRightFunc(ss[i], unicode.IsSpace)
+	}
+	return ss
+}
+
+func modelPackageName(api *api.API, packageNameOverride string) string {
+	if len(packageNameOverride) > 0 {
+		return packageNameOverride
+	}
+	return api.Name
+}
+
+func validatePackageName(newPackage, elementName, sourceSpecificationPackageName string) error {
+	if sourceSpecificationPackageName == newPackage {
+		return nil
+	}
+	// Special exceptions for mixin services
+	if newPackage == "google.cloud.location" ||
+		newPackage == "google.iam.v1" ||
+		newPackage == "google.longrunning" {
+		return nil
+	}
+	if sourceSpecificationPackageName == newPackage {
+		return nil
+	}
+	return fmt.Errorf("rust codec requires all top-level elements to be in the same package want=%s, got=%s for %s",
+		sourceSpecificationPackageName, newPackage, elementName)
+}
+
+func validateModel(api *api.API, sourceSpecificationPackageName string) error {
+	// Set the source package. We should always take the first service registered
+	// as the source package. Services with mixes will register those after the
+	// source package.
+	if len(api.Services) > 0 {
+		sourceSpecificationPackageName = api.Services[0].Package
+	} else if len(api.Messages) > 0 {
+		sourceSpecificationPackageName = api.Messages[0].Package
+	}
+	for _, s := range api.Services {
+		if err := validatePackageName(s.Package, s.ID, sourceSpecificationPackageName); err != nil {
+			return err
+		}
+	}
+	for _, s := range api.Messages {
+		if err := validatePackageName(s.Package, s.ID, sourceSpecificationPackageName); err != nil {
+			return err
+		}
+	}
+	for _, s := range api.Enums {
+		if err := validatePackageName(s.Package, s.ID, sourceSpecificationPackageName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func imports(importMap map[string]*goImport) []string {
+	var imports []string
+	for _, imp := range importMap {
+		imports = append(imports, fmt.Sprintf("%q", imp.path))
+	}
+	return imports
+}
+
+func generateMethod(m *api.Method) bool {
+	// Ignore methods without HTTP annotations, we cannot generate working
+	// RPCs for them.
+	// TODO(#499) - switch to explicitly excluding such functions. Easier to
+	//     find them and fix them that way.
+	return !m.ClientSideStreaming && !m.ServerSideStreaming && m.PathInfo != nil && len(m.PathInfo.PathTemplate) != 0
+}
+
+// The list of Golang keywords and reserved words can be found at:
+//
+// https://go.dev/ref/spec#Keywords
```

Comment:
This should be replaced with the list of Dart reserved words.  Maybe just make it an empty list for now?